src/hal/src/pso/descriptor.rs

```diff
@@ -1,4 +1,16 @@
 //! Descriptor sets and layouts.
+//!
+//! Resource descriptors are used by shaders for accessing resources like buffers
+//! and images. The use of a descriptor has three stages:
+//!
+//! - Descriptor layout is specified during pipeline creation
+//! - Descriptor set is allocated from a descriptor pool
+//! - Descriptor set is bound during rendering
+//!
+//! A Descriptor Set Layout describes the types of resources to be used by a pipeline.
+//!
+//! A Descriptor Set contains the actual resources to be bound/used: eg. buffer, images
```

Comment:
Descriptor sets are actually views into resource, so an indirection. Basically a pointer to a memory slice and how it's supposed to be accessed by the shader.